scalalib/src/mill/scalalib/bsp/ScalaMetalsSupport.scala

```diff
@@ -3,23 +3,44 @@ package mill.scalalib.bsp
 import mill.api.experimental
 import mill.{Agg, T}
 import mill.define.Target
+import mill.scalalib.api.ZincWorkerUtil.isScala3
 import mill.scalalib.{Dep, DepSyntax, ScalaModule}
+import mill.api.Result
 
 /*+ Enable some common settings required to properly support Metals Language Server (via BSP). */
 @experimental
 trait ScalaMetalsSupport extends ScalaModule {
 
   /** The semanticDB version to use. It needs to support your configured Scala versions. */
-  def semanticDbVersion: T[String]
+  def semanticDbVersion: T[String] = T { "" }
+
   override def scalacPluginIvyDeps: Target[Agg[Dep]] = T {
-    super.scalacPluginIvyDeps() ++ Agg(
-      ivy"org.scalameta:::semanticdb-scalac:${semanticDbVersion()}"
-    )
+    if (!isScala3(scalaVersion()) && semanticDbVersion().isEmpty) {
+      val msg =
+        """|
+           |When using ScalaMetalsSupport with Scala 2 you must provide a semanticDbVersion
+           |
+           |def semanticDbVersion = ???
+           |""".stripMargin
+      Result.Failure(msg)
+    } else {
+      Result.Success(
+        super.scalacPluginIvyDeps() ++ Agg(
+          ivy"org.scalameta:::semanticdb-scalac:${semanticDbVersion()}"
+        )
+      )
+    }
```

Comment:
Looks like we're missing the happy Scala 3 case now, which does not need to add any dep at all.

Reply:
ah, whoops, fixed.